main.rb

```diff
@@ -0,0 +1,94 @@
+module Enumerable
+  def my_each
+    index = 0
+
+    while index < size
+      yield self[index]
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    item = 0
+    index = 0
+
+    while index < size
+      yield self[item], index
+      index += 1
+      item += 1
+    end
+  end
+
+  def my_select
+    arr = []
+
+    my_each do |item|
+      arr.push(item) if yield item
+    end
+    arr
+  end
+
+  def my_all?
+    bool = true
+
+    my_each do |item|
+      bool = false unless yield item
+    end
+    bool
+  end
+
+  def my_any?
+    bool = false
+
+    my_each do |item|
+      bool = true if yield item
+    end
+    bool
+  end
+
+  def my_none?
+    bool = true
+
+    my_each do |item|
+      bool = false if yield item
+    end
+    bool
+  end
+
+  def my_count
+    count = 0
+
+    my_each do |item|
+      count += 1 if yield item
+    end
+    count
+  end
+
+  def my_map
+    arr = []
+
+    my_each do |item|
+      arr.push(yield item)
+    end
+    arr
+  end
```

Comment:
- [ ] should have working `my_map` **(covered by automated tests)** identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method should return an enumerator if no block or argument is given.